airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/stream_slicer.py

```diff
@@ -23,24 +22,9 @@ class StreamSlicer(RequestOptionsProvider):
     """
 
     @abstractmethod
-    def stream_slices(self, sync_mode: SyncMode, stream_state: StreamState) -> Iterable[StreamSlice]:
+    def stream_slices(self) -> Iterable[StreamSlice]:
```

Comment:
After [discussing this internally](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1686939006489719), we decided to remove the sync_mode from the interface of this method